### ISSUE TITLE
Regression for macro, maybe intended, low priority

### ISSUE THREAD

**Issue by StefanSalewski** [nim-lang/Nim#11038]
```nim
import strutils

#const ct4nt = [
let ct4nt = [
  "int32|int",
  "uint32|int",
  "gboolean|toBool",
  "cstring|$$"]

proc ct5nt(s: string): string =
  for i in ct4nt:
    if i.startsWith(s):
      return i[s.len + 1 .. ^1]

macro myMacro(arg: untyped): untyped =
 
  var h = ct5nt("test")


myMacro("Hallo")
```

```
t.nim(17, 7) Hint: 'h' is declared but not used [XDeclaredButNotUsed]
t.nim(20, 8) template/generic instantiation of `myMacro` from here
t.nim(4, 5) Error: cannot generate VM code for ct4nt = ["int32|int", "uint32|int", "gboolean|toBool", "cstring|$$"]

```

Using const instead of let fixes it, so it is not really a bug. Unfortunately took me some time to discover that simple fix.

Reference:

https://github.com/StefanSalewski/gintro/blob/master/gintro/gimpl.nim

https://github.com/StefanSalewski/gintro/issues/42

**Comment by StefanSalewski** [comment]
OK, will close it as it is an intended regression and error message is fine. gintro is already fixed in latest release 0.4.17

**Comment by krux02** [comment]
yes, you are accessing a runtime value at compile time. This is supposed to not compile.
